src/App.tsx

```diff
@@ -1,14 +1,43 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { filterTodos } from './utils/filterTodos';
+import { getTodoById } from './utils/getTodoById';
+import { TodoCompletedCategory } from './types/todoCompletedCategory';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [electTodoId, setElectTodoId] = useState<number>(0);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [query, setQuery] = useState<string>('');
+  const [todoCategory, setTodoCategory] = useState<TodoCompletedCategory>(
+    TodoCompletedCategory.all,
+  );
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .finally(() => {
+        setVisibleTodos(todos);
+        setLoading(false);
+      });
+  }, []);
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.